src/charm.py

```diff
@@ -87,6 +92,115 @@ def _spring_boot_port(self) -> int:
             return port
         return 8080
 
+    def _exec(
+        self, command: typing.List[str], environment: typing.Optional[typing.Dict[str, str]] = None
+    ) -> typing.Tuple[int, str, str]:
```

Comment:
Would it be better to use a `typing.NamedTuple` with the exit_code, stdout and stderr attributes? That way it is easier for consumers to know how to access the data?

Reply:
Added, thanks!